adb.h

```diff
@@ -157,7 +157,6 @@ typedef struct adb_client_s {
 } adb_client_t;
 
 typedef struct adb_context_s {
-    adb_client_t *clients;
```

Comment:
The idea behind tracking all the clients is it's then possible to terminate properly the server, releasing all the resources. This is unfortunately not implemented yet so there is an `adb_register_client()` function without `adb_UNregister_client()`.

It's fine for now to remove it as the feature is not implemented. In this case `client->next` becomes useless too.

Reply:
@spiriou rebase to fix the conflict.

Reply:
Can you remove the `next` field too in `adb.h:struct adb_client`?
```
142 typedef struct adb_client_s {
143    struct adb_client_s *next;
```

Reply:
Done.